### PR TITLE
multi: create dedicated page for adding wallet

### PR DESCRIPTION
**Pull request by oshorefueled**
### Resolves

Issue #75 

### What's new
This PR:
* adds a dedicated page for accessing the create and restore functionalities.
* It also adds the page as the default page when no wallet exists. 
* adds a button on the wallets page to access the page
* removes the create and restore buttons in the header

### Relevant screenshots or logs
Screenshot 2020-04-21 at 8 03 49 PM

Screenshot 2020-04-21 at 8 04 19 PM

Screenshot 2020-04-21 at 8 04 31 PM

Screenshot 2020-04-21 at 8 04 45 PM

### CODE REVIEW

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
image

the input fields and the suggestion buttons are not aligned.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
image

this looks like you used the password modal instead of a page with password input fields. Any resaon why?

**Review by Sirmorrison — COMMENTED** [review]
image

Do we really need the input field title label? we can set the `isTitleLabel = false` for the input fields. reason for this is the `...` after the `Input word 12`

**Comment by oshorefueled** [comment]
> image
> 
> this looks like you used the password modal instead of a page with password input fields. Any resaon why?

I didn't use a password modal, I used a modal with input fields for the password and confirm password fields. There's no reason to have them on a separate page as the layout is slim and it follows the normal password input convention of other pages.

File: ui/create_restore_page.go [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+type createRestore struct {
+	gtx          *layout.Context
+	theme        *decredmaterial.Theme
+	info         *wallet.MultiWalletInfo
+	wal          *wallet.Wallet
+	keyEvent     chan *key.Event
+	errChan      chan error
+	showRestore  bool
+	restoring    bool
+	showPassword bool
```
Comment (Sirmorrison) [inline ui/create_restore_page.go:43]: I think you meant `showCreate`

File: ui/create_restore_page.go [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+								pg.Restore()()
+							} else {
+								pg.mainContent()()
+							}
+						}),
+					)
+				})
+			}),
+		)
+		if pg.showPassword {
```
Comment (Sirmorrison) [inline ui/create_restore_page.go:155]: This is supposed to show the create modal i believe. it should be `showCreat` instead.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

**Comment by oshorefueled** [comment]
> Do

the label should denote what number of seed is being inputted, since they have to be done in an order. You can check the mobile mockups to see how it was done there.  It can however be simplified to look exactly like that.

**Comment by Sirmorrison** [comment]
> > Do
> 
> the label should denote what number of seed is being inputted, since they have to be done in an order. You can check the mobile mockups to see how it was done there. It can however be simplified to look exactly like that.

cool. Then lets remove the `...`

File: ui/create_restore_page.go [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+			}),
+			layout.Flexed(1, func() {
+				layout.Center.Layout(pg.gtx, func() {
+					layout.Flex{}.Layout(pg.gtx,
+						layout.Rigid(func() {
+							pg.gtx.Constraints.Width.Max = pg.gtx.Constraints.Width.Max / 2
+							pg.inputsGroup(pg.seedListLeft, 16, 0)
+						}),
+						layout.Rigid(func() {
+							//fmt.Printf("max %v min %v \n", pg.gtx.Constraints.Width.Max, pg.gtx.Constraints.Width.Min)
```
Comment (Sirmorrison) [inline ui/create_restore_page.go]: If this is not used, lets remove it. no need keeping commented codes.

File: ui/create_restore_page.go [hunk 4]
```diff
@@ -0,0 +1,555 @@
...
+
+	"github.com/raedahgroup/dcrlibwallet"
+	"github.com/raedahgroup/godcr/wallet"
+
+	"gioui.org/layout"
+	"gioui.org/text"
+	"gioui.org/unit"
+	"gioui.org/widget"
+	"github.com/raedahgroup/godcr/ui/decredmaterial"
+	"golang.org/x/exp/shiny/materialdesign/icons"
```
Comment (Sirmorrison) [inline ui/create_restore_page.go:18]: On a lighter note, could we try rearranging the imports? standard imports at the top, other imports below?. I think it makes our work cleaner and more orderly.

Comment (oshorefueled) [reply ui/create_restore_page.go:18]: > On a lighter note, could we try rearranging the imports? standard imports at the top, other imports below?. I think it makes our work cleaner and more orderly.

doesn't really matter. As long as it builds on the CI, it's fine.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
image

* After a wallet restore seed has been successfully add, the password mode that shows up has a title of `create wallet` and the button at the bottom say `restore wallet`. I think this is misleading. Kindly have the modal title read `Restore Wallet`.

* Also when a user navigates out of the restore page, the input fields should be cleared.

**Review by Sirmorrison — APPROVED** [review]
PR looks good.

**Review by beansgum — APPROVED** [review]

**Pull request merged**